Assets/Project/Core/Actors/Obstacles/ObstacleSpawner.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.InteropServices.WindowsRuntime;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+public class ObstacleSpawner : MonoBehaviour, ISpawner
+{
+    //TODO: Implement spawnRate (currently the obstacles reposition themselves using
+    // Lean Tween OnComplete event
+    //TODO: Test creating a range on Y for spawning
+    //TODO: SpawingData must be configurable on a separate data class
+    [SerializeField] private GameObject obstacleToSpawn;
+    [SerializeField] private float spawnRate;
+    [SerializeField] private int spawnLimit;
+    [SerializeField] private int objectSpeed;
+    [SerializeField] private float movementRange;
+
+    [SerializeField] private readonly List<Obstacle> spawnedObstacles = new List<Obstacle>();
+    public bool activated;
```

Comment:
Faz sentido essa variável ser pública? Ela é usada fora da classe? Se for, ela precisa ser setada?

Sugestão: colocar como privada ou criar uma propriedade dela

Reply:
Era pra uma abordagem diferente, esqueci de tirar! Valeu pelo feedback!